com.unity.render-pipelines.universal/ShaderLibrary/Lighting.hlsl

```diff
@@ -585,7 +605,7 @@ half4 UniversalFragmentPBR(InputData inputData, half3 albedo, half metallic, hal
     BRDFData brdfData;
     InitializeBRDFData(albedo, metallic, specular, smoothness, alpha, brdfData);
     
-    Light mainLight = GetMainLight(inputData.shadowCoord);
+    Light mainLight = GetMainLight(inputData.positionWS, inputData.shadowCoord);
```

Comment:
I wonder if from API point of view if it wouldn't be better to add a function to apply shadow fade instead of implicitly computing it in the GetMainLight and add the extra overrides. 

```suggestion
    Light mainLight = GetMainLight(inputData.shadowCoord);
    mainLight.shadowAttenuation = ApplyShadowFade(mainLight.shadowAttenuation, inputData.positionWS)
```

@ellioman what do you think?

Reply:
Yes, I agree. It's a lot clearer with your suggestion.

I would also like the `GetMainLight(shadowCoord)` function to have a clearer name like `GetMainLightWithRealtimeShadow(shadowCoord)` but that's a problem for another day :)

Reply:
I think your solution looks cleaner

Reply:
Fixed in latest commit